File: C++/Dynamic Programming/Fibonacci Sequence/FibonacciSequence.cpp
#include "algorithm-visualizer.h"
#include <iostream>
#include <string>

Array1DTracer tracer("Sequence");

void InitFibonacciSequence()
{
	Layout::setRoot(VerticalLayout({ tracer }));
}

int main()
{
	InitFibonacciSequence();
	
	int index = 15;
	int D[15] = { 1, 1 };
	for (int i = 2; i < index; i++) {
		D[i] = 0;
	}
	tracer.set(D);
	Tracer::delay();

	for (int i = 2; i < index; i++) {
		D[i] = D[i - 2] + D[i - 1];
		tracer.select(i - 2, i - 1);
		Tracer::delay();
		tracer.patch(i, D[i]);
		Tracer::delay();
		tracer.depatch(i);
		tracer.deselect(i - 2, i - 1);
	}
}

File: Dynamic Programming/Fibonacci Sequence/FibonacciSequence.java
import org.algorithm_visualizer.*;
import java.util.*;

public class Main {
	
	Array1DTracer tracer = new Array1DTracer("Sequence");
	
	Main() {
		Layout.setRoot(new VerticalLayout(new Commander[] { tracer }));
		int index = 15;
		ArrayList<Integer> D = new ArrayList<Integer>(index);
		D.add(1);
		D.add(1);
		for (int i = 2; i < index; i++) {
		  D.add(0);
		}
		tracer.set(D);
		Tracer.delay();

		for (int i = 2; i < index; i++) {
		  D.add(i, D.get(i-2) + D.get(i-1));
		  tracer.select(i - 2, i - 1);
		  Tracer.delay();
		  tracer.patch(i, D.get(i));
		  Tracer.delay();
		  tracer.depatch(i);
		  tracer.deselect(i - 2, i - 1);
		}
	}
	
	public static void main(String[] args) {
		new Main();
	}
}
